app/environments/brassbirmingham/brassbirmingham/envs/classes/player.py

```diff
@@ -235,87 +268,111 @@ def canDevelop(self, building1: Building, building2: Building) -> bool:
             and not building2.isRetired
             and building2.canBeDeveloped
             and building2.owner == self
+            and self.isCardInHand(discard)
         )
 
     # 4 SELL
-    def canSell(self, building: MarketBuilding) -> bool:
+    def canSell(self, building: MarketBuilding, discard: Card) -> bool:
         return (
             building.isActive
             and building.owner == self
             and self.canAffordSellBuilding(building)
+            and self.isCardInHand(discard)
         )
 
     # 5 LOAN
-    def canLoan(self) -> bool:
-        return self.income >= 3
+    def canLoan(self, discard) -> bool:
+        return self.incomeLevel() >= 3 and self.isCardInHand(discard)
 
     # 6 SCOUT
-    def canScout(self, additionalDiscard: Card) -> bool:
-        ownership = False
+    def canScout(self, discard: Card, additionalDiscards: List[Card]) -> bool:
+        assert len(additionalDiscards) == 2
+
+        if not self.isCardInHand(discard):
+            return False
+        for card in additionalDiscards:
+            if not self.isCardInHand(card):
+                return False
+
         for card in self.hand.cards:
             if card.name in [CardName.wild_location, CardName.wild_industry]:
```

Comment:
Should we check the wild location/industry decks and if they're empty?

Reply:
In a legal game, they can't be empty and those other statements be true